metals/src/main/scala/scala/meta/internal/metals/debug/DebugProxy.scala

```diff
@@ -95,6 +107,32 @@ private[debug] final class DebugProxy(
         .map(DebugProtocol.syntheticResponse(request, _))
         .foreach(client.consume)
 
+    case request @ CompletionRequest(args) =>
+      val completions = for {
+        frame <- lastFrames.find(_.getId() == args.getFrameId())
+        originalSource = frame.getSource()
+        sourceUri = clientAdapter.toMetalsPath(originalSource.getPath)
+      } yield {
+        compilers.completions(
+          sourceUri,
+          new Position(frame.getLine() - 1, 0),
+          EmptyCancelToken,
+          args
+        )
+      }
+      completions
+        .getOrElse(Future.failed(new Exception("No source data available")))
+        .map { items =>
+          val responseArgs = new CompletionsResponse()
+          responseArgs.setTargets(items.toArray)
+          val response = new DebugResponseMessage
+          response.setId(request.getId)
+          response.setMethod(request.getMethod)
+          response.setResult(responseArgs.toJson)
+          client.consume(response)
+        }
+        .withTimeout(15, TimeUnit.SECONDS)
```

Comment:
Shouldn't timeout be smaller? 15 seconds seems like a lot.

Reply:
Changed to 5, should be enough.